pkg/domain/filter.go

```diff
@@ -0,0 +1,26 @@
+package domain
+
+import (
+	"fmt"
+)
+
+// ConfigurationItem contains the meaningful elements from the AWS Config Configuration Item
+// (https://docs.aws.amazon.com/config/latest/developerguide/config-item-table.html) that are
+// required for filtering.
+type ConfigurationItem struct {
+	ResourceType string `json:"ResourceType"`
```

Comment:
I believe ResourceType should be resourceType:
https://docs.aws.amazon.com/config/latest/developerguide/example-sns-notification.html
![image](https://user-images.githubusercontent.com/3370907/54616246-66c9dd80-4a2d-11e9-9cad-673921c827ed.png)

Additionally, the Go SDK provides a ConfigurationItem model you could use. This would also help keep json tags our of our domain package (if that is still a guideline we want to follow). While there are _some_ inconsistencies, the resourceType field is all we care about right now, and that appears to be consistent.

re: Inconsistencies -- I talked to our TAM, and he asked that I open a support ticket:
https://console.aws.amazon.com/support/cases?region=us-east-1#/5898018761/en

Reply:
The problem with using the `ConfigurationItem` model from the SDK is the *actual* AWS output has the value for the `configuration` key as a JSON object, but the SDK expects that value to be a string, which causes the unmarshal to fail.

https://docs.aws.amazon.com/config/latest/developerguide/example-sns-notification.html
<img width="641" alt="Screen Shot 2019-03-19 at 10 23 23 AM" src="https://user-images.githubusercontent.com/21367055/54618766-8a8f2280-4a31-11e9-85d3-d854721d9990.png">

https://docs.aws.amazon.com/sdk-for-go/api/service/configservice/#ConfigurationItem
<img width="724" alt="Screen Shot 2019-03-19 at 10 26 25 AM" src="https://user-images.githubusercontent.com/21367055/54618739-819e5100-4a31-11e9-998e-45ef8b749590.png">

Reply:
I have added that to my support ticket. The json tag should still be `resourceType` instead of `ResourceType` i believe

Reply:
My bad, this should be fixed now.